staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -106,6 +108,73 @@ func ParsePrivateKeyPEM(keyData []byte) (interface{}, error) {
 	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
 }
 
+// ParsePublicKeysPEM is a helper function for reading an array of rsa.PublicKey or ecdsa.PublicKey from a PEM-encoded byte array.
+// Reads public keys from both public keys, private keys, and certificate files.
+func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
+	ok := false
+	var keys []interface{}
+	for len(keyData) > 0 {
+		var block *pem.Block
+		block, keyData = pem.Decode(keyData)
+		if block == nil {
+			break
+		}
+
+		switch {
```

Comment:
I'm still concerned about breaking currently working clusters by strictly enforcing PEM headers (specifically ones mixing up `RSA PRIVATE KEY`/`PRIVATE KEY` and `RSA PUBLIC KEY`/`PUBLIC KEY` PEM headers).

Is there a good reason not to simply try to parse the forms we recognize from the PEM blocks, as we do today?